inject-java/src/test/groovy/io/micronaut/inject/factory/nullreturn/NullReturnFactorySpec.groovy

```diff
@@ -83,22 +80,20 @@ class Test2 {}
         cs.size() == 2
         cs.any { it.name == "one" }
         cs.any { it.name == "two" }
-        factory.bCalls == 5
-        factory.cCalls == 3
 
         expect: "1 is created for D"
         beanContext.getBeansOfType(D).size() == 1
         beanContext.getBean(D, Qualifiers.byName("one"))
-        factory.bCalls == 6
-        factory.cCalls == 4
         factory.dCalls == 2 //2 C beans
 
         and: "1 is created for D2"
-        beanContext.getBeansOfType(D2).size() == 1
+        beanContext.getBeansOfType(D2).size() == 1 // D two is disabled
```

Comment:
Changes in `AbstractInitializableBeanDefinition` where `argument.isDeclaredNullabl` is added brought those changes, but I think it's the correct behaviour